twitterbot.py

```diff
@@ -1,36 +1,67 @@
+import csv
 import os
 
-import pandas as pd
 import tweepy
+import click
 
-df = pd.read_csv('followers.csv', index_col=False)
-list_users_url = list(df['twitter_url'])
 
+@click.command()
+@click.option('--consumer_key', default=os.environ.get('CONSUMER_KEY'),
+              help='CONSUMER_KEY')
+@click.option('--consumer_secret', default=os.environ.get('CONSUMER_SECRET'),
+              help='CONSUMER_KEY')
+@click.option('--access_token', default=os.environ.get('ACCESS_TOKEN'),
+              help='ACCESS_TOKEN')
+@click.option('--access_token_secret',
+              default=os.environ.get('ACCESS_TOKEN_SECRET'),
+              help='ACCESS_TOKEN_SECRET')
+@click.option('--limit', default=1000,
+              help='set limit amount of users you want to follow')
+def main(consumer_key, consumer_secret, access_token,
+         access_token_secret, limit):
+    with open('followers.txt', 'r') as file:
+        list_users_url = []
+        reader = csv.reader(file)
+        for row in reader:
+            list_users_url.append(row[0])
 
-list_users = []
+    list_users = []
+    for user_url in list_users_url:
+        if 'https://twitter.com/' in user_url:
+            list_users.append(user_url.split('https://twitter.com/')[-1])
+        elif 'http://twitter.com/' in user_url:
+            list_users.append(user_url.split('http://twitter.com/')[-1])
 
-for user_url in list_users_url:
-    list_users.append(user_url.split('https://twitter.com/')[-1])
+    auth = tweepy.OAuthHandler(consumer_key, consumer_secret)
+    auth.set_access_token(access_token, access_token_secret)
 
-consumer_key = os.environ.get('CONSUMER_KEY')
-consumer_secret = os.environ.get('CONSUMER_SECRET')
-access_token = os.environ.get('ACCESS_TOKEN')
-access_token_secret = os.environ.get('ACCESS_TOKEN_SECRET')
+    api = tweepy.API(
+        auth, wait_on_rate_limit=True,
+        wait_on_rate_limit_notify=True,
+        retry_count=10, retry_delay=5,
+        retry_errors=5
+    )
+    print(api)
+    counter = 0
+    with open('followers_cache.txt', 'ar+') as f:
+        cached_users = []
+        tmp_cache = []
+        reader = csv.reader(f)
+        for row in reader:
+            cached_users.append(row[0])
+        for user in list_users:
+            if ('https://twitter.com/' + user) not in cached_users and (
```

Comment:
this line with bug, first of all
- cached_users must be set(), 
- in `followers.txt` can contain both `http://` or `https://`

Reply:
"followers.txt can contain both http:// or https://"
Nope, in our case we talking about "followers_cache.txt" and it contains only "https://", because we writting it down here ```f.write('https://twitter.com/' + user + '\n')```

Reply:
but i find one way of little optimization of this function, i'll commit it asap.